### PR TITLE
Stojanovic/ep 41 file upload download support

### PR DESCRIPTION
**Pull request by Ducica**

### CODE REVIEW

**Comment by Ducica** [comment]
@mirekys this one too please!

File: ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/components/FileUploader/FileUploaderWrappers.jsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+export const FileUploadWrapper = ({
+  preactComponent,
+  uploadWrapperClassName,
+  uploadButtonClassName,
+  props,
+}) => {
+  const preactCompRef = useRef();
+  useEffect(() => {
+    render(
```
Comment (mirekys) [inline ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/components/FileUploader/FileUploaderWrappers.jsx:14]: This Uppy usage looks really wild, hard to maintain & understand. I think we should actually be using existing React components & hooks https://uppy.io/docs/react/ to implement the uppy uploader dialog, are there any blockers preventing us from using it (@corovcam)? From docs, it looks like any of the `preact` magic shouldn't be needed here :)

File: ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/deposit/NRDocumentValidationSchema.js [hunk 2]
```diff
@@ -39,10 +36,7 @@ export const unique = (value, context, path, errorString) => {
 export const NRDocumentValidationSchema = Yup.object().shape({
   metadata: Yup.object().shape({
     // not sure but I assume it would be good idea to ask for a minimum length of title
```
Comment (mirekys) [inline ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/deposit/NRDocumentValidationSchema.js:38]: This comment could also be removed after the consensus that we cannot enforce any title length on users, other than that atleast some value is required

Comment (Ducica) [reply ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/deposit/NRDocumentValidationSchema.js:38]: @mirekys for sure, my bad

File: ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/components/FileUploader/FileUploaderWrappers.jsx (hunk 1 above)
Comment (corovcam) [reply ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/components/FileUploader/FileUploaderWrappers.jsx:14]: There's one big issue. When building custom Uppy plugins, rendered inside Uppy Dashboard, Uppy uses Preact components and hooks. Meaning, Preact is bundled inside the package and there's no alternative (preact/compat doesn't work). The Preact hooks are incompatible with React, so if React calls Preact hook (in custom plugin) in repository, error is issued.

**Review by mirekys — APPROVED** [review]

**Review by mirekys — APPROVED** [review]

**Pull request merged**